common/filesystem.h

```diff
@@ -1237,6 +1451,13 @@ inline auto Internal::FileRefBase::WriteableDestroy() -> void {
       "calling `Close` and handling any errors to avoid data loss.");
 }
 
+template <OpenAccess A>
+auto FileRef<A>::Truncate(int64_t new_size) -> ErrorOr<Success, FdError>
+  requires Writeable
```

Comment:
Aside: I wonder if `public Internal::FileRefBase` should be `private` inheritance since it would let you bypass these `Writable` checks and get surprises; we would never want anyone to ever upcast.